### PR TITLE
add german language locale support

### PR DESCRIPTION
**Pull request by so2liu**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/3599)

### CODE REVIEW

File: components/date-picker/locale/de_DE.tsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import DatePickerLocale from "rmc-date-picker/lib/locale/en_US";
+
+export default {
+  okText: "OK",
+  dismissText: "Abrechen",
```
Comment (moritzjacobs) [inline components/date-picker/locale/de_DE.tsx:5]: Typo :)

```suggestion
  dismissText: "Abbrechen",
```

File: components/date-picker/locale/de_DE.tsx [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  extra: "bitte auswählen",
```
Comment (moritzjacobs) [inline components/date-picker/locale/de_DE.tsx:6]: Should be capitalized.

```suggestion
  extra: "Bitte auswählen",
```

File: components/input-item/locale/de_DE.tsx [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export default {
+  confirmLabel: "Fertig",
+  backspaceLabel: "Zurück",
+  cancelKeyboardLabel: "TastaturAbrechen",
```
Comment (moritzjacobs) [inline components/input-item/locale/de_DE.tsx:4]:
```suggestion
  cancelKeyboardLabel: "TastaturAbbrechen",
```

File: components/menu/locale/de_DE.tsx [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export default {
+  okText: "Ok",
+  cancelText: "Abrechen",
```
Comment (moritzjacobs) [inline components/menu/locale/de_DE.tsx:3]:
```suggestion
  okText: "OK",
  cancelText: "Abbrechen",
```

File: components/picker/locale/de_DE.tsx [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export default {
+  okText: "Ok",
+  dismissText: "Abrechen",
+  extra: "bitte auswählen",
```
Comment (moritzjacobs) [inline components/picker/locale/de_DE.tsx:4]:
```suggestion
  okText: "OK",
  dismissText: "Abbrechen",
  extra: "Bitte auswählen",
```

File: components/pull-to-refresh/locale/de_DE.tsx [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export default {
+  activateText: "Aktualisieren sofort nach der Freigabe",
+  deactivateText: "Ziehen nach unten zum Aktualisieren",
+  finishText: "Aktualisieren fertig",
+};
```
Comment (moritzjacobs) [inline components/pull-to-refresh/locale/de_DE.tsx:5]:
```suggestion
  activateText: "Nach freigabe sofort aktualisieren",
  deactivateText: "Zum aktualisieren nach unten ziehen",
  finishText: "Aktualisieren beendet",
};
```

File: components/search-bar/locale/de_DE.tsx [hunk 7]
```diff
@@ -0,0 +1,3 @@
+export default {
+  cancelText: "Abrechen",
```
Comment (moritzjacobs) [inline components/search-bar/locale/de_DE.tsx:2]:
```suggestion
  cancelText: "Abbrechen",
```

**Comment by awmleer** [comment]
v2 已经不再维护了，这个 PR 先关闭了哈，如果有兴趣的话可以关注一下最新的 v5 版本～

**Pull request closed without merging**